e2e_playwright/st_tabs_selection_test.py

```diff
@@ -87,9 +87,26 @@ def test_maintains_selection_when_same_name_exists(
     wait_for_app_run(app)
     # Add Tab 3
     control_buttons.nth(0).click()
+    wait_for_app_run(app)
     # Select Tab 2
     tab_buttons = app.get_by_test_id("stTabs").locator("button[role=tab]")
     tab_buttons.nth(1).click()
+
+    # Ensure that the click worked and the highlight animation finished
+    # to avoid issues with the snapshot later
+    expect(tab_buttons.nth(1)).to_have_attribute("aria-selected", "true")
+    tab_highlight_element = app.query_selector(
+        "[data-baseweb='tab-highlight']", strict=True
+    )
+    assert tab_highlight_element is not None
```

Comment:
nit: are you sure that this won't create any flakiness? Wouldn't it be better to add another:

```
expect(app.locator("[data-baseweb='tab-highlight']")).to_be_visible()
```

Reply:
ok, in that case its fine .. but maybe add it as a comment on top of the assert

Reply:
I changed it 🙂 but will add the comment anyways